File: libraryProject_Python/Book.py
from Media import Media

class Book(Media):

	def __init__(self,callNo,title,subject,author,description,publisher,city,year,series,notes):
		self.callNo = callNo
		self.title = title
		self.subject = subject
		self.author = author
		self.description = description
		self.publisher = publisher
		self.city = city
		self.year = year
		self.series = series
		self.notes = notes
		super(Book,self).__init__(self.callNo,self.title,self.subject,self.notes)


	
	def compare_Other(self,other):
		if other in self.description:
			return True
		elif other in self.year:
			return True
		elif other in self.notes:
			return True
		else:
			return False




	def display(self):
		super(Book,self).display()
		print ("Author: %s" % (self.author))
		print ("Description: %s" % (self.description))
		print ("Publisher: %s" % (self.publisher))
		print ("City: %s" % (self.city))
		print ("Year: %s" % (self.year))
		print ("Series: %s\n\n" % (self.series))
	#	print "Notes: %s" % self.notes
		

File: LibraryProject_C++/Book.cpp
#include "Book.h"

Book::Book(const string& call_no,const string& title,const string& subjects,const string& author,const string& description,const string& publisher,const string& city,const string& year,const string& series,const string& notes):Media(call_no,title,subjects,notes),author(author), description(description), publisher(publisher)
,city(city), year(year), series(series)
{
}


Book::~Book(){}

/*
   compare_other finds if the input string exists in Description, Notes
   and year. It returns boolean value.
 */
bool Book::compare_other(const string& other_string) const{

	if(description.find(other_string)!=string::npos){
		return true;
	}
	else if(year.find(other_string)!=string::npos){
		return true;
	}
	else if(Media::compare_other(other_string)){
		return true;
	}
	else{
		return false;
	}
}

/*
   display function to print book attributes. This function gives
   call to display() of Media class.
 */
void Book::display() const
{
	cout << "------Book------" << endl;

	Media::display();  
	cout<<"Author:      "<<author<<endl;
	cout<<"Description: "<<description<<endl;
	cout<<"Publisher:   "<<publisher<<endl;
	cout<<"City:        "<<city<<endl;
	cout<<"Year:        "<<year<<endl;
	cout<<"Series:      "<<series<<endl<<endl;


}

